### ISSUE TITLE
loss not decreasing with deepspeed

### ISSUE THREAD

**Issue by haorannlp** [Stability-AI/StableLM#71]
I'm trying to finetune the model with the base-3b using my own data on 8GPUs with deepspeed stage 3 and gradient ckpting:

The ds config looks like follows:

```{
    "zero_optimization": {
        "stage": 3,
        "offload_optimizer": {
            "device": "cpu",
            "pin_memory": true
        },
        "offload_param": {
            "device": "cpu",
            "pin_memory": true
        },
        "overlap_comm": true,
        "contiguous_gradients": true,
        "sub_group_size": 0,
        "reduce_bucket_size": "auto",
        "stage3_prefetch_bucket_size": "auto",
        "stage3_param_persistence_threshold": "auto",
        "stage3_max_live_parameters": 0,
        "stage3_max_reuse_distance": 0,
        "stage3_gather_16bit_weights_on_model_save": true
    },
    "fp16": {
        "enabled": true,
        "auto_cast": false,
        "loss_scale": 0,
        "initial_scale_power": 16,
        "loss_scale_window": 1000,
        "hysteresis": 2,
        "min_loss_scale": 1
    },
    "optimizer": {
        "type": "AdamW",
        "params": {
          "lr": "auto",
          "betas": [
            0.9,
            0.999
          ],
          "eps": 1e-8,
          "weight_decay": 0
        }
    },
    "train_batch_size": "auto",
    "train_micro_batch_size_per_gpu": "auto",
    "wall_clock_breakdown": false
}```

If I do not use deepspeed on a single GPU, the loss decreases during training; with  deepspeed on single GPU, the loss still decreases but looks slower; on 8gpu, the loss cannot decrease. Any suggestions? Thanks~

**Comment by rSemology** [comment]
There is something wrong with deepspeed at the moment.
